Shared/Samples/Add ENC exchange set/AddENCExchangeSetView.swift

```diff
@@ -0,0 +1,160 @@
+                        self.error = error
+                    }
+                }
+        }
+        .errorAlert(presentingError: $error)
+    }
+}
+
+private extension AddENCExchangeSetView {
+    class Model: ObservableObject {
+        /// A map with Oceans style.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISOceans)
+            map.initialViewpoint = Viewpoint(
+                latitude: -32.5,
+                longitude: 60.95,
+                scale: 67000
+            )
+            return map
+        }()
+        
+        /// The geometry that represents a rectangular shape that encompasses the area on the
+        /// map where the ENC data will be rendering.
+        private(set) var completeExtent: Envelope?
+        
+        /// A URL to the temporary SENC data directory.
+        private let temporaryURL: URL = {
+            let directoryURL = FileManager.default.temporaryDirectory.appendingPathComponent(
+                ProcessInfo().globallyUniqueString
+            )
+            // Create and return the full, unique URL to the temporary folder.
+            try? FileManager.default.createDirectory(
+                at: directoryURL,
+                withIntermediateDirectories: true
+            )
+            return directoryURL
+        }()
+        
+        deinit {
+            // Recursively remove all files in the sample-specific
+            // temporary folder and the folder itself.
+            try? FileManager.default.removeItem(at: temporaryURL)
+            // Reset ENC environment display settings.
+            let displaySettings = ENCEnvironmentSettings.shared.displaySettings
+            displaySettings.textGroupVisibilitySettings.resetToDefaults()
+            displaySettings.viewingGroupSettings.resetToDefaults()
+        }
+        
+        /// Gets the ENC exchange set data and loads it and sets the display settings.
+        func addENCExchangeSet() async throws {
+            let exchangeSet = ENCExchangeSet(fileURLs: [.exchangeSet])
+            // URL to the "hydrography" data folder that contains the "S57DataDictionary.xml" file.
+            let resourceURL = URL.hydrographyDirectory.deletingLastPathComponent()
+            // Set environment settings for loading the dataset.
+            let environmentSettings = ENCEnvironmentSettings.shared
+            environmentSettings.resourceURL = resourceURL
+            // The SENC data directory is for temporarily storing generated files.
+            environmentSettings.sencDataURL = temporaryURL
+            updateDisplaySettings()
+            try await exchangeSet.load()
+            try await renderENCData(dataset: exchangeSet.datasets)
+        }
+        
+        /// Maps the exchange set data to ENC layer and ENC cells and loads the layers.
+        /// - Parameter dataset: The ENC dataset previously loaded.
+        private func renderENCData(dataset: [ENCDataset]) async throws {
```

Comment:
```suggestion
        /// - Parameter datasets: The ENC datasets previously loaded.
        private func renderENCData(datasets: [ENCDataset]) async throws {
```

Sorry I didn't realize this was an array the first time around.